clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -255,6 +255,8 @@ def err_drv_expecting_fopenmp_with_fopenmp_targets : Error<
   "The option -fopenmp-targets must be used in conjunction with a -fopenmp option compatible with offloading, please use -fopenmp=libomp or -fopenmp=libiomp5.">;
 def err_drv_expecting_fsycl_with_sycl_opt : Error<
   "The option %0 must be used in conjunction with -fsycl to enable offloading.">;
+def err_drv_fsycl_with_c_type : Error<
+  "The option %0%1 must not be used in conjunction with -fsycl which expects C++ source.">;
```

Comment:
How would the driver behave on simply receiving a c source/header without `-x c++` for SYCL compilation?
Unless this scenario is already covered, it seems that a more general diagnostic would only mention the input type, and be triggered in both the `.c` and `-x c` cases. Aka:
`The %0 file type must not be compiled with -fsycl which expects C++ source.`

Reply:
I tested it already by passing a c/cheader without -x option to clang++, it doesn't force c compilation on the c/c-header based on the type of the file.  Instead I got the same output as if it is cpp source file.

Reply:
Here is output from the tool that suggests the .c file is compiled as if it is .cpp. So, the behavior of the tool when -x(c/c-header) is as expected. I think this diagnostic should only address user specifically forcing c-compilation.  

clang-11: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]